addon/@ember/test-waiters/wait-for.ts

```diff
@@ -80,6 +117,8 @@ function wrapFunction(fn: Function, label?: string) {
 
     if (isThenable(result)) {
       return waitForPromise(result, label);
+    } else if (isGenerator(result)) {
```

Comment:
I think technically you don't want `isGenerator` check here, you want to check if `result` conforms to the iterable protocol (does it have a `done` and a `next` function).

Reply:
As stated above, I think we do want to look for a generator object here. We could potentially relax it to look for anything implementing the iterator protocol (which I'm pretty sure only has a `next` method), but as stated (further) above, I'm a little uncomfortable with being that loose when we don't know of any uses cases that require us to handle iterators-but-not-generators.

Reply:
This looks correct. We decided to keep this as is.